### DISCUSSION TITLE
Creating project clones should copy patch files

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#232] [Feature Requests]
Recently I have been working on a series of projects that all require a similar setup. Every time I "save as," none of my patch files are actually saved to the new patch. Ideally they would copy over to the new patch, but I suppose a workaround could be to add a button to download a zip of all patch files so they are easier to re-upload.

**Comment by tastez** [comment]
IMHO its better to just upload your project files somewhere else and then reference them in your multiple copies of projects. Duplicating files seems wasteful - but what sort of files do you need to copy over?

**Reply by positlabs** [reply]
Well in this case it's not even that I need to keep those files, but the result is a broken project after saving it. I don't think that's ideal.

Mainly I'm thinking of the future. I want to be able to build out some templates and have my coworkers copy them to use as starting points for their projects.

I could upload them somewhere else but that gets a little tedious.

**Reply by steam0r** [reply]
all reference to files in cables are plain urls, if you copy your project you simply copy these references. that's intended. we are thinking about a "collect assets" button to move these over to the current project at least.

saving the project should not result in a broken state, though. how did you get to that?

**Reply by positlabs** [reply]
Ok, I see that it does maintain the asset reference if you use a texture or something. 

I was using `op.patch.getAssetPath()` to get the asset path dynamically. When it looks for those assets in the copied project, it doesn't exist because the assets aren't transferred.

**Comment by tgd-git** [comment]
i would even expand this problem to copy pasting ops from one patch to an other. maybe there could be a way to leave references default behaviour but make copying files optional. "collect assets" button sounds great btw.
i am facing some inconvenience right now working collaboratively on a bigger project. lets say you want to prepare a little module for the project and you do it in a separate patch. the module uses a ton of assets. now there is no way i could integrate this module into the project without reuploading and relinking the assets. depending on the amount of assets and iterative changes to the module this can be a real pain.
